corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -92,18 +94,24 @@ def update_celery_state(sender=None, headers=None, **kwargs):
 )
 def prune_synclogs():
     """
-    Drops all partition tables containing data that's older than 63 days (7 weeks)
+    Drops all partition tables containing data that's older than 63 days (9 weeks)
     """
-    oldest_synclog = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
-    while oldest_synclog and (datetime.today() - oldest_synclog).days > SYNCLOG_RETENTION_DAYS:
+    db = router.db_for_write(SyncLogSQL)
+    oldest_date = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
+    while oldest_date and (datetime.today() - oldest_date).days > SYNCLOG_RETENTION_DAYS:
         year, week, _ = oldest_synclog.isocalendar()
         table_name = "{base_name}_y{year}w{week}".format(
             base_name=SyncLogSQL._meta.db_table,
             year=year,
             week="%02d" % week
         )
         drop_query = "DROP TABLE IF EXISTS {}".format(table_name)
-        db = router.db_for_write(SyncLogSQL)
         with connections[db].cursor() as cursor:
             cursor.execute(drop_query)
-        oldest_synclog = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
+        oldest_date += timedelta(weeks=1)
```

Comment:
If there's an orphaned synclog, that'll just stay and round and all future cleanup will start at that date that's frozen in time. So as time progresses each successive cleanup step will have more and more weeks to go through.

I guess if we also have a way to clean up those old orphaned synclogs then that's not an issue, I'm just curious what you're thinking for that. One option is to just literally after the rest of the cleanup delete all synclogs older than the cutoff date, which after table-dropping cleanup will only be a small number of orphaned rows, if anything.

Reply:
Yeah, I mentioned that in the PR description, it will add the extra week to the task, but that will be a miniscule amonut of time (`DROP IF EXISTS` for a table that does not will be quick). I think we can clean those up manually, but leaving them around at first, while notifiying datadog gives us a chance to investigate before we do.

Reply:
Oh sorry, yup. Isee that now!